test/src/matomo_event_test.dart

```diff
@@ -377,5 +458,20 @@ void main() {
 
       expect(matomotoMap['ca'], null);
     });
+
+    test('it should not have performanceInfo if its a ping', () async {
+      final matomotoMap = getCompleteMatomoEvent()
+          .copyWith(
+            ping: true,
+          )
+          .toMap();
+
+      expect(matomotoMap['pf_net'], null);
+      expect(matomotoMap['pf_srv'], null);
+      expect(matomotoMap['pf_tfr'], null);
+      expect(matomotoMap['pf_dm1'], null);
+      expect(matomotoMap['pf_dm2'], null);
+      expect(matomotoMap['pf_onl'], null);
```

Comment:
Prefer using the `isNull` matcher.

Reply:
Now thinking about it, `containsKey()` might be even better than `[]` since I want to test if the key is present at all, while `[]` could return `null` if the key is present (and set to `null`).